core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3296,7 +3296,7 @@ object Stream extends StreamLowPriority {
   def fromEither[F[_]]: PartiallyAppliedFromEither[F] =
     new PartiallyAppliedFromEither(dummy = true)
 
-  private[fs2] final class PartiallyAppliedFromIterator[F[_]](
+  final class PartiallyAppliedFromIterator[F[_]](
```

Comment:
Because this class was package-private, so we received no bincompat support from MiMa. Indeed, the fact that this works at all seems like a manifestation of https://github.com/scala/bug/issues/6794.